src/generators/sitemap/index.mjs

```diff
@@ -0,0 +1,81 @@
+import { readFile, writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+
+import dedent from 'dedent';
+
+import { BASE_URL } from '../../constants.mjs';
+
+/**
+ * This generator generates a sitemap.xml file for search engine optimization
+ *
+ * @typedef {Array<ApiDocMetadataEntry>} Input
+ *
+ * @type {GeneratorMetadata<Input, string>}
+ */
+export default {
+  name: 'sitemap',
+
+  version: '1.0.0',
+
+  description: 'Generates a sitemap.xml file for search engine optimization',
+
+  dependsOn: 'metadata',
+
+  /**
+   * Generates a sitemap.xml file
+   *
+   * @param {Input} entries
+   * @param {Partial<GeneratorOptions>} options
+   * @returns {Promise<string>}
+   */
+  async generate(entries, { output }) {
+    const lastmod = new Date().toISOString().split('T')[0];
+
+    const apiPages = entries
+      .filter(entry => entry.heading.depth === 1)
+      .map(entry => {
+        const path = entry.api_doc_source.replace(/^doc\//, '/docs/latest/');
+        const url = new URL(path, BASE_URL).href;
+
+        return {
+          loc: url,
+          lastmod,
+          changefreq: 'weekly',
+          priority: '0.8',
+        };
+      });
+
+    apiPages.push({
+      loc: new URL('/docs/latest/api/', BASE_URL).href,
+      lastmod,
+      changefreq: 'daily',
+      priority: '1.0',
+    });
+
+    const template = await readFile(
+      join(import.meta.dirname, 'template.xml'),
+      'utf-8'
+    );
+
+    const urlset = apiPages
+      .map(
+        page => dedent`
+        <url>
```

Comment:
nit: use template files for this and then do simple key->value substitution. Or use proper rss/feed libraries OR xml libraries.

Reply:
I'm fine with whatever that uses the least amount of dependencies. We can also just use yet another template file, we can also simply use another dependency, like the xast one, or rss/feeds or whatever.